src/suricata.c

```diff
@@ -1498,7 +1498,7 @@ static TmEcode ParseCommandLine(int argc, char** argv, SCInstance *suri)
         {"af-packet", optional_argument, 0, 0},
         {"netmap", optional_argument, 0, 0},
         {"pcap", optional_argument, 0, 0},
-        {"pcap-file-continuous", 0, 0, 0},
+        {"pcap-file-continuous", optional_argument, 0, 0},
```

Comment:
This doesn't make much sense to me. This value gets getopt that a argument to 'pcap-file-continuous' is optional. It's not, there is no option.

With master:
<pre>
 ./src/suricata -c suricata.yaml -l tmp/ -r ~/pcaps/ --pcap-file-continuous=blah --disable-detection
./src/suricata: option '--pcap-file-continuous' doesn't allow an argument
</pre>
This branch:
<pre>
./src/suricata -c suricata.yaml -l tmp/ -r ~/pcaps/ --pcap-file-continuous=blah --disable-detection
[95908] 12/12/2017 -- 17:41:24 - (suricata.c:1764) <Info> (ParseCommandLine) -- detection engine disabled
</pre>
The master behavior for this is correct.

However, I do see that this change somehow fixes the other option parsing. It's not a proper fix though, and probably masks the real issue.